### DISCUSSION TITLE
Taro v3.6 的路由库应用场景是什么样的？

### DISCUSSION THREAD

**Discussion by knight-peter** [NervJS/taro#13502] [Q & A] [answered]
最初，得知taro支持vue-router等路由库的时候，还是很开心的。后来其实是在其中一个page上实现的单页应用，他们之间的跳转不会让小程序增加回退按钮，意思是没有增加小程序的路由历史。
这样的话，就有点不知道该在那种场景下使用了。
希望有人能来解答下。

**Comment by AdvancedCat** [comment, marked as answer]
感谢反馈。

小程序本身是 MPA 架构的，它提供一套属于自己的路由系统，开发者使用 `navigateTo` 或 `redirectTo` 等 api 操作原生页面的跳转。在执行这个 api 的背后，小程序维护了页面堆栈，也可能有其他内部操作，对外界是透明的。
Taro 的目的是提供一套开发者熟悉的 DSL 开发小程序应用，但在编译时仍会输出为小程序要求的应用架构，自然也是遵守的这套路由系统。因此，小程序是**规范**，Taro是解决问题的**手段**。

回到你的问题，如果要做成整个应用的路由系统，那么势必需要接管小程序路由规范，既要符合开发者在Web开发的认知，又要处理好小程序路由中的差异。一旦做的不好，或者失控，对于两边都是不好的体验，“不守规矩，不成方圆”。所以，页面间的跳转仍是需要遵守小程序的规范，至于页面内就可以“为所欲为”了。

所以，一开始我们的目的不是要做成整个应用的路由管理，而是在运行时中引入 `location` 和 `history` api，前端路由库恰好在它两的基础上可以在小程序中跑起来。即使不用前端路由库，你也可以使用 `location`  `history` 的组合实现一个简易路由系统。这也是它的应用场景了。

如果，你有更好的想法，也可以进一步讨论。

最后，引用 Taro 老大哥的一段话：
https://github.com/NervJS/taro-rfcs/pull/3#issuecomment-630626561
